packages/api/src/useCases/cases/lockAndFetchCase.ts

```diff
@@ -0,0 +1,39 @@
+import type { User } from "@moj-bichard7/common/types/User"
+import type { FastifyBaseLogger } from "fastify"
+import type postgres from "postgres"
+
+import type { AuditLogDynamoGateway } from "../../services/gateways/dynamo"
+import type DataStoreGateway from "../../services/gateways/interfaces/dataStoreGateway"
+import type { ApiAuditLogEvent } from "../../types/AuditLogEvent"
+
+import createAuditLogEvents from "../createAuditLogEvents"
+import { lockExceptions } from "./lockExceptions"
+
+export const lockAndFetchCase = async (
+  dataStore: DataStoreGateway,
+  auditLogGateway: AuditLogDynamoGateway,
+  caseId: number,
+  user: User,
+  logger?: FastifyBaseLogger
+) => {
+  await dataStore
+    .transaction(async (callbackSql: postgres.Sql) => {
+      const auditLogEvents: ApiAuditLogEvent[] = []
+
+      const caseMessageId = (await dataStore.selectCaseMessageId(caseId)).message_id
+
+      await lockExceptions(dataStore, callbackSql, caseId, user, auditLogEvents)
+
+      // TODO: Check permissions for Triggers
+      // TODO: Lock Triggers and AuditLog them
+
+      if (auditLogEvents.length > 0) {
+        createAuditLogEvents(auditLogEvents, caseMessageId, auditLogGateway, logger)
```

Comment:
This can return an error, but it doesn't throw. What happens to the transaction if there is an error creating an audit log event?

Reply:
It's the same as in the UI (I think). 

The transaction only applies to the DB. I couldn't get the transaction to rollback with an error thrown in this function.

Reply:
I missed the `await` 🤦‍♂️